src/RRTStar/RRTStarWrapper.cpp

```diff
@@ -0,0 +1,167 @@
+#include "src/RRTStar/rrt_star.h"
+
+#include <algorithm>
+#include <eigen3/Eigen/Dense>
+#include <map>
+
+double SPACEDIM = 2.0;
+double PI = 3.14159;
+
+extern "C" {
+    // RRT* algorithm
+    //
+    // Arguments:
+    //      xStart: starting x position
+    //      yStart: starting y position
+    //      xEnd: ending x position
+    //      yEnd: ending y position
+    //      stepSize: sampling distance
+    //      maxIterations: maximum number of iterations to take
+    //      obstacles: list of obstacles formatted as
+    //          [lower_left_x, lower_left_y, upper_right_x, upper_right_y]
+    //      numObstacles: number of obstacles
+    // Returns:
+    //      xPath: list to store x path
+    //      yPath: list to store y path
+    int ApplyRRTStar(double xStart, double yStart, double xEnd, double yEnd,
+            int stepSize, int maxIterations, double* obstacles_llx,
+            double* obstacles_lly, double* obstacles_urx, double* obstacles_ury,
+            int numObstacles, double* xPath, double* yPath)
+    {
+        RRT* rrt = new RRT;
+
+        // Initialize configuration
+        rrt->reset();
+        rrt->setStateSpace(xStart, yStart, xEnd, yEnd);
+        rrt->setStepSize(stepSize);
+        rrt->setMaxIterations(maxIterations);
+        rrt->initialize();
+
+        // Construct obstacles
+        for (int i = 0; i < numObstacles; i++) {
+            rrt->addObstacle(
+                    Vector2f(obstacles_llx[i], obstacles_lly[i]),
+                    Vector2f(obstacles_urx[i], obstacles_ury[i])
+            );
+        }
+
+        // Declare variables
+        Node *qBest, *qNew, *qNearest;
+        double cost, dist, distBest, gamma, radius;
+        int reached = 0;
+
+        // Cost to each vertex
+        map<Node *, double> cost_map;
+        cost_map[rrt->root] = 0;
+
+        // Run RRT*
+        gamma = 1 + pow(2, SPACEDIM) * (1 + 1 / SPACEDIM) *
+                rrt->getFreeArea();
+        for (int i = 0; i < rrt->max_iter; i++) {
+            qNew = rrt->getRandomNode();
+            qNearest = rrt->nearest(qNew->position);
+            if (rrt->distance(qNew->position, qNearest->position) >
+                rrt->step_size) {
+                Vector2f newConfig = rrt->newConfig(qNew, qNearest);
+                if (rrt->isSegmentInObstacles(newConfig, qNearest->position)) {
+                    continue;
+                }
+                qNew = new Node;
```

Comment:
You can have a struct constructor. IMHO, this should take position because you always set it after you create a Node object.

Reply:
Done